src/cpp/src/partition_manager.cpp

```diff
@@ -145,9 +189,19 @@ shared_ptr<ModifyTimingInfo> PartitionManager::add(
     if (!vectors.defined() || !vector_ids.defined()) {
         throw runtime_error("[PartitionManager] add: vectors or vector_ids is undefined.");
     }
+
     if (vectors.size(0) != vector_ids.size(0)) {
         throw runtime_error("[PartitionManager] add: mismatch in vectors.size(0) and vector_ids.size(0).");
     }
+
+    if(attributes_table!=nullptr && attributes_table->num_rows()!= vector_ids.size(0)){
+        throw runtime_error("[PartitionManager] add: mismatch in attributes_table and vector_ids size.");
+    }
+
+    if(attributes_table!=nullptr && !attributes_table->GetColumnByName("id")){
```

Comment:
Ideally, we shouldn't keep track of ids in the arrow table since we already keep track of them in the index partitions.